BrainPortal/app/models/cluster_task.rb

```diff
@@ -747,6 +747,7 @@ def setup_and_submit_job
             self.status_transition(self.status, "Failed To Setup")
           else
             self.addlog("Setup and submit process successful.")
+            self.meta[:setup_time] = Time.now
```

Comment:
You are changing the main CBRAIN code base. It is not proper to modify framework-level code that way.

The point of having special extensions and modules is that they provide their functionality INDEPENDENTLY from the rest of the framework, without requiring changes in that framework. The recording of the :setup_time must be made by code in the module.

Reply:
ok, so you do not care of same file being used by an old style tool and this new boutiques integrator based tool?

Extensions is not even in plugin folder.

Reply:
What make no sense? Do you think a file used by old type of task cannot be used by a new type of boutiques task? Or perhaps this even is improbable... ok than

I understand that's easy to override just thought the 'good design' solution is  inferior.

Reply:
It makes no sense to record the timestamp for the setup() method for tasks that are not integrated with this module.

THis timestamps is not associated with any input files. It is associated with the task's lifecycle.

Reply:
The point is that this feature here is meant to be configured for tasks that use some inputs file while at the same task we expect no other tasks to use those same input files. And if any task is launched AFTER the current task and synchronize an input we use in the current task, then we do not delete the input. That's it.